### PR TITLE
Feature: Initial assortment of refactor changes for review

### LINKED ISSUE #3342: Feature: Add the ability to close the primary/left panel
**Requirements**
- Add support for closing the left pane

### PR DESCRIPTION
**Pull request by lukeblevins**
TBD

Closes #3342

(Obviously, do not merge these)

### CODE REVIEW

**Comment by QuaintMako** [comment]
Cannot give huge useful feedback on the refactoring as a whole aside that it looks very promising.

The work on regrouping the different commands in `IItemCommandService.cs` is a very good idea. 

Been only playing with commands around the recycle bin, one can feel there is a lot of convoluted code around creating new commands. Something here begs to be streamlined.

Even if the behaviors are slightly different depending from where we're coming (Sidebar / Former IShellPage for exemple), maybe it'd be possible to create a common base for each command? This refactor gives a small starting ground for it at least.

File: src/Files.App/Filesystem/FilesystemOperations/FilesystemOperations.cs [hunk 1]
```diff
@@ -1,996 +1,776 @@
...
-                        progress?.Report(100.0f);
-                        errorCode?.Report(FileSystemStatusCode.InProgress | FileSystemStatusCode.Success);
+                        return await CopyItemsAsync(source, destination, collisions, progress, errorCode, cancellationToken);
                     }
-                    else
+                }
+                else if (copyResult.Items.Any(x => CopyEngineResult.Convert(x.HResult) == FileSystemStatusCode.InUse))
+                {
+                    var failedSources = copyResult.Items.Where(x => CopyEngineResult.Convert(x.HResult) == FileSystemStatusCode.InUse);
+                    var filePath = failedSources.Select(x => x.HResult == CopyEngineResult.COPYENGINE_E_SHARING_VIOLATION_SRC ? x.Source : x.Destination);
```
Comment (QuaintMako) [inline src/Files.App/Filesystem/FilesystemOperations/FilesystemOperations.cs:129]: This expression is used several times through the file. Wrapping it up in a method would be worth.

File: src/Files.App/Filesystem/FilesystemOperations/FilesystemOperations.cs [hunk 2]
```diff
@@ -1,996 +1,776 @@
...
+            var sourceRename = sourceNoSkip.Zip(collisionsNoSkip, (src, coll) => new { src, coll }).Where(item => item.coll != FileNameConflictResolveOptionType.ReplaceExisting).Select(item => item.src);
+            var destinationRename = destinationNoSkip.Zip(collisionsNoSkip, (src, coll) => new { src, coll }).Where(item => item.coll != FileNameConflictResolveOptionType.ReplaceExisting).Select(item => item.src);
 
-                errorCode?.Report(fsResult);
-                if (item != null)
+            var result = (FilesystemResult)true;
+            var copyResult = new ShellOperationResult();
+            if (sourceRename.Any())
+            {
+                var (status, response) = await connection.SendMessageForResponseAsync(new ValueSet()
```
Comment (QuaintMako) [inline src/Files.App/Filesystem/FilesystemOperations/FilesystemOperations.cs:70]: Several of those constructs are repeated in the code. Maybe a method to regroup their creation into one method taking arguments to customize the message in Entry could be worth the trouble to set up?

**Review by QuaintMako — COMMENTED** [review]
Made a few comments here on there, mostly on `FileSystemOperation.cs` 

Overall, all the changes you made are going into the right directly I feel like! The use of `working directory` instead of the `associatedInstance` is gonna simplify so much worked.

If it doesn't exist, a WorkingDirectoryHelper or something akin would be a great addition!

**Comment by QuaintMako** [comment]
Looks like my code selection for the review has been a bit messy... Hope you'll be able to make something out of it @lukeblevins.

If some modifications I suggested look good to you but you don't have time for them, hit me up and I'll see what I can do!

First code review for me as well on Git, if you've got any feedback, I'll take it!

File: src/Files.App/Filesystem/FilesystemOperations/FilesystemOperations.cs [hunk 3]
```diff
@@ -1,996 +1,776 @@
...
                         {
-                            if (NativeFileOperationsHelper.HasFileAttribute(source.Path, FileAttributes.Hidden))
+                            (status, response) = await connection.SendMessageForResponseAsync(new ValueSet()
                             {
-                                // The source folder was hidden, apply hidden attribute to destination
-                                NativeFileOperationsHelper.SetFileAttribute(fsCopyResult.Result.Path, FileAttributes.Hidden);
-                            }
-                            copiedItem = (BaseStorageFolder)fsCopyResult;
+                                { "Arguments", "FileOperation" },
+                                { "fileop", "CreateFile" },
```
[the remark this answers is not in the document]
Comment (lukeblevins) [reply src/Files.App/Filesystem/FilesystemOperations/FilesystemOperations.cs:182]: The keywords will go away soon once we drop the FullTrust component :)

**Comment by QuaintMako** [comment]
I think the #10185 is partially adressed in this refactoring.

**Pull request closed without merging**